test/http-client.spec.js

```diff
@@ -787,20 +784,49 @@ describe('http client', () => {
         });
     });
 
-    it('should be applied after all transformers', (done) => {
-      class RequestInterceptor {
+    it('should be possible to modify a header in a request interceptor', (done) => {
+      class Interceptor {
         request(message) {
+          message.headers.add('aheader', 'anothervalue');
+          return message;
+        }
+
+        response(message) {
+          var request = jasmine.Ajax.requests.mostRecent();
+          expect(request.requestHeaders['aheader']).toBe('anothervalue')
```

Comment:
This test will pass if the client fails to execute the interceptor. I think we could move this assertion into the `.then` continuation on line 812.